incubator/hnc/internal/validators/object.go

```diff
@@ -34,10 +35,11 @@ const (
 // +kubebuilder:webhook:path=/validate-objects,mutating=false,failurePolicy=ignore,groups="*",resources="*",verbs=create;update;delete,versions="*",name=objects.hnc.x-k8s.io
 
 type Object struct {
-	Log     logr.Logger
-	Forest  *forest.Forest
-	client  client.Client
```

Comment:
You should be able to use this client instead of the raw K8s client, which means you don't have to import anything new or add any of that extra plumbing.

Reply:
Updated in latest commit